airflow/providers/google/cloud/hooks/compute_ssh.py

```diff
@@ -303,9 +387,29 @@ def _authorize_os_login(self, pubkey):
         self.log.info("Importing SSH public key using OSLogin: user=%s", username)
         expiration = int((time.time() + self.expire_time) * 1000000)
         ssh_public_key = {"key": pubkey, "expiration_time_usec": expiration}
-        response = self._oslogin_hook.import_ssh_public_key(
-            user=username, ssh_public_key=ssh_public_key, project_id=self.project_id
-        )
+        if self._check_running_in_parallel():
+            self.log.info("Several processes are executed in parallel")
+            bucket_name = f"tmp_bucket_{self.project_id}"
+            lock_file_name = "file.lock"
+            lock_path = f"gs://{bucket_name}/{lock_file_name}"
+            self._create_bucket_for_lock(bucket_name=bucket_name)
```

Comment:
What about if we move these lines:
```python
bucket_name = f"tmp_bucket_{self.project_id}"
lock_file_name = "file.lock"
lock_path = f"gs://{bucket_name}/{lock_file_name}"
```
inside the method
```python
self._create_bucket_for_lock()
```
and rename it something like `def _create_lock() -> str`?
In this case we wouldn't have to repeat the bucket name construction (DRY Principle) and encapsulate the lock implementation, so it would be easier to modify it in the future.

Reply:
It is a good idea to move variables creation to the method, but speaking about this method *self._create_bucket_for_lock()* - it is not responsible for creating the lock. Instead, it only prepares a bucket to handle this lock, the lock is created later in *_acquire_lock()* method.

Reply:
After refactoring this initialization code is not repeated anymore, thus I have nothing against it. Thank you!